### DISCUSSION TITLE
Type casting bound query string values

### DISCUSSION THREAD

**Discussion by bgurney** [livewire/livewire#3908] [1. Help]
I have a property bound to the query string that receives a large integer value (think long ID number or reference).

When the component properties are hydrated from the query string values, I'd like to specify a type cast for that value so that I can ensure it's treated as a string, not an integer. The reason I need to do this is that large integers go out of range in the JavaScript environment and end up getting rounded.

### Example

Take this URL: `https://demo.test/?ref=6634511510158295630`

This is what I'm having to do at the moment.

```php
public function mount()
{
    $this->ref = (string) $this->ref;
}
```

Is there a better way to do this?

**Comment by denizozturk** [comment]
I use it this way in similar situations, this is the current usage method for php. (`(string), (int), (float), (bool)`...)
The alternative method is to do this with functions, (`strval,intval,floatval,boolval`...)

**Reply by bgurney** [reply]
Thanks for your reply. I'm happy with how to cast in PHP, I'm curious as to whether there's a more declarative way to cast the properties on hydration, rather than having to reset the property in the `mount()` method. I have tried using PHP 8 property casts but I get an error, so assume that syntax isn't supported in Livewire yet...

**Reply by denizozturk** [reply]
I want to be sure, you're getting this `ref` from the request, and your point is to confirm that it's a string in all the places it's going to be used? Are you trying to make sure of that?

By the way, all of the examples I gave above are methods that can be used in the current livewire version.

**Reply by bgurney** [reply]
Yes I basically want to cast the `ref` property on the component. I have just seen that property casters used to exist in Livewire v1 but were removed in v2.

https://laravel-livewire.com/docs/2.x/upgrading#casters

I think, based on what I've read here, what I've done is the recommended approach for now... Hopefully support for casting to primitives will come back now there is an actual PHP syntax for it.

**Reply by denizozturk** [reply]
You don't need to wait, you need to do it with livewire hydrate. You can do this with hydrate in the new version, you can try it. Did you try this?

```
public function hydrateRef($value)
{
    $this->foo = (string) $value; // strval($value);
}

public function dehydrateRef($value)
{
    $this->foo = (string) $value; // strval($value);
}
```

**Reply by bgurney** [reply]
Thanks for the suggestion of using hydrate methods. I was aware of them and will look into them further. Still not the declarative syntax I was hoping for, but hopefully that will come.

**Comment by joshhanley** [comment]
@bgurney what you have done, is the best for casting the query string for now.

But there is a PR #3679 in the works, which will allow you to hook into the query string hydration/dehydration to perform any transformations like casts, so if that gets merged, that would be where I'd do it.

Hope this helps!

**Reply by bgurney** [reply]
Cool, I'll keep my eye on this - sounds like a nice hook point addition!
